File: 1642-water-bottles/water-bottles.c
int replace(int numBottles, int numExchange, int emptyBottle) {
    
    printf("%d %d %d\n", numBottles, numExchange ,emptyBottle);
    if (emptyBottle >= numExchange) {
        numBottles += emptyBottle / numExchange;
        emptyBottle = emptyBottle % numExchange;
    }

    if (numBottles  == 0) {
        return 0;
    }

    return numBottles +   replace(numBottles/numExchange  , numExchange, emptyBottle + (numBottles % numExchange) );
}

int numWaterBottles(int numBottles, int numExchange) {
    return replace(numBottles, numExchange, 0);
}

File: 1642-water-bottles/water-bottles.py
class Solution:
    def numWaterBottles(self, numBottles: int, numExchange: int) -> int:
        res = numBottles
        leftover = 0
        while(numBottles >= numExchange):
            leftover = numBottles % numExchange
            numBottles = numBottles // numExchange
            res += numBottles
            numBottles += leftover
        return res